packages/libs/src/sdk/services/Solana/programs/ClaimableTokens.ts

```diff
@@ -0,0 +1,196 @@
+            blockhash,
+            lastValidBlockHeight
+          }
+        }
+      })
+      return { userBank, didExist: false }
+    }
+    return { userBank, didExist: true }
+  }
+
+  /**
+   * Creates a claimable tokens program transfer instruction using configured
+   * program ID, mint addresses, derived nonce, and derived authorities.
+   *
+   * Must be paired with a matching Secp256k1 instruction.
+   * @see {@link createTransferSecpInstruction}
+   */
+  async createTransferInstruction(params: CreateTransferRequest) {
+    const {
+      feePayer: feePayerOverride,
+      ethWallet,
+      mint,
+      destination
+    } = await parseParams(
+      'createTransferInstruction',
+      CreateTransferSchema
+    )(params)
+    const feePayer = feePayerOverride ?? (await this.solana.getFeePayer())
+    const source = await this.deriveUserBank({ ethWallet, mint })
+    const nonceKey = ClaimableTokensProgram.deriveNonce({
+      ethAddress: ethWallet,
+      authority: this.authorities[mint],
+      programId: this.programId
+    })
+    return ClaimableTokensProgram.createTransferInstruction({
+      payer: feePayer,
+      sourceEthAddress: ethWallet,
+      sourceUserBank: source,
+      destination,
+      nonceAccount: nonceKey,
+      authority: this.authorities[mint],
+      programId: this.programId
+    })
+  }
+
+  /**
+   * Creates a signed Secp256k1 instruction for a claimable tokens transfer
+   * using configured program ID, derived nonce, and derived authorities.
+   *
+   * @see {@link createTransferInstruction}
+   */
+  async createTransferSecpInstruction(params: CreateSecpRequest) {
+    const { ethWallet, destination, amount, mint, instructionIndex, auth } =
+      await parseParams(
+        'createTransferSecpInstruction',
+        CreateSecpSchema
+      )(params)
+
+    let nonce = BigInt(0)
+    const nonceKey = ClaimableTokensProgram.deriveNonce({
+      ethAddress: ethWallet,
+      authority: this.authorities[mint],
+      programId: this.programId
+    })
+    const nonceAccount = await this.connection.getAccountInfo(nonceKey)
+    const encodedNonceData = nonceAccount?.data
+    if (encodedNonceData) {
+      const nonceData =
+        ClaimableTokensProgram.layouts.nonceAccountData.decode(encodedNonceData)
+      nonce = nonceData.nonce
+    }
+    const data = ClaimableTokensProgram.createSignedTransferInstructionData({
+      destination,
+      amount: wAUDIO(amount).value,
+      nonce
+    })
+    const [signature, recoveryId] = await auth.sign(data)
+    return Secp256k1Program.createInstructionWithEthAddress({
+      ethAddress: ethWallet,
+      message: data,
+      signature,
+      recoveryId,
+      instructionIndex
+    })
+  }
+
+  /**
+   * Derives the user bank of a user from their Ethereum wallet and the token mint.
+   *
+   * Use {@link getOrCreateUserBank} instead if you want to ensure the userBank exists.
+   */
+  public async deriveUserBank(params: DeriveUserBankRequest) {
```

Comment:
I don't love the pattern of "method in sdk calls method in spl verbatim" because it feels like how libs used to have /api and /services and a tonnnnn of code dupe.
but I can't propose anything better atm, and this feels pretty reasonable

Reply:
Yeah I don't love it, and really any point where we're deriving a user bank we should probably be creating one... but likewise couldn't think of it better